runtime/README.md

```diff
@@ -106,56 +108,60 @@ The contexts contain the internal structures necessary to resume operations to s
+The encrypted CMKs have the structure:
 
-### CMIDs
+| **Name**      | **Bits** | **Description**                |
+| ------------- | -------- | ------------------------------ |
+| padding       | 32       | reserved                       |
+| iv            | 96       |                                |
+| ciphertext    | 768      | encrypted CMK data (see above) |
+| GCM tag       | 128      |                                |
 
-CMIDs offload storage of key metadata to mailbox callers. They therefore impose no additional storage within Caliptra Core.
+The total size of the CMK is therefore 128 bytes.
 
-Unecrypted CMIDs have the following structure:
+Only the encrypted CMKs will appear in mailbox messages.
 
-| **Name**     | **Bits** | **Description**                             |
-| ------------ | -------- | ------------------------------------------- |
-| padding      | 80       | 0                                           |
-| block offset | 24       | Starting block number                       |
-| size         | 24       | Size (in bytes) of the data stored.         |
-|              |          | Does not have to be a multiple of 16 bytes. |
+The key used to encrypt the CMKs is randomized on reset, which means that CMKS cannot be used between resets.
+The iv is a randomized 1-up counter that is incremented for every key created.
 
-Hence, data for a single CMID must be stored contiguously.
+#### Key Usage
 
-Padding is kept for additional future features.
+The internal CMK structure and several commands use a key usage tag to specify how a key can be used:
 
-The CMIDs are AES-256-GCM encrypted, with a 128-bit ciphertext concatenated with the 128-bit AES-GCM tag.
+| **Value**  | **Usage**   |
+| ---------- | ----------- |
+| 0          | Reserved    |
+| 1          | HMAC        |
+| 2          | HKDF        |
+| 3          | AES-256-GCM |
 
-### Storage Keys
+#### Replay Prevention and Deletion
 
-The key for the data referenced by a CMID is unique and derived via HKDF from the CDI<sub>RT</sub> with the info being the concatenation of `"CMID" || block offset || size`.
+To prevent replay attacks, Caliptra will have a small table that maps a CMK's internal ID to its last known usage counters.
+Whenever a CMK is used, this table is checked and updated.
 
-The IV is the concatenation of `block offset || size`.
+This is necessary for AES-256-GCM in particular to ensure that keys are only used a certain number of times, as per [NIST SP 800-38D, Section 8.3](https://doi.org/10.6028/NIST.SP.800-38D).
+Only AES-256-GCM keys need to be tracked in this table.
 
-This scheme avoids having to store the IV while avoiding IV reuse problems.
-
-GCM prevents an adversary from changing internal CMID parameters, such as changing an AES key to an ECDSA key, which has sensitive internal structure (although ECDSA is not currently supported).
+This requires 96 bits of storage per AES-256-GCM key. These can stored as a sorted list to the DCCM requirements.
```

Comment:
I'd be in favor of keeping it more vague here on the implementation detail. It might depend a little on testing and implementation. (Keeping it sorted makes the most sense to me right now and would probably be the most compact.)